c/bpxskt.c

```diff
@@ -369,8 +372,32 @@ Socket *tcpClient4(SocketAddress *socketAddress,
       /* manual says returnCode and value only meaningful if return value = -1 */
       *returnCode = 0;
       *reasonCode = 0;
+
+      if (readTimeoutInMillis > 0 && writeTimeoutInMillis <= 0) {
+        writeTimeoutInMillis = readTimeoutInMillis;
+        if (socketTrace){
```

Comment:
I think you can. I didn't suggest they were related, it's just one usually wants to adjust both.

I'd do something like this:
```c
readTimeoutInMillis = readTimeoutInMillis >= 0 ? readTimeoutInMillis : DEFAULT_RECV_TIMEOUT_MS;
writeTimeoutInMillis = writeTimeoutInMillis >= 0 ? writeTimeoutInMillis : DEFAULT_SEND_TIMEOUT_MS;
```

Reply:
i am putting this limitation here whether or not the code is capable, because it seems weird to not set both.

Reply:
Not sure if this is weird. In one of our product, we have a non-zero default timeout for receive operations but a zero default timeout (i.e., the call block indefinitely) for send operations. The justification, I think, is that we can tolerate a "slow" peer when we send, but once sent, we expect a relatively quick response because a slow response would indicate issues/incompatibility in our case. Timeout which depend on each other look unintuitive.